### PR TITLE
Menu bar load image 

### PR DESCRIPTION
**Pull request by 1killerbehindyou1**
Merge with master

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,5 @@ Mentoring QT project.codeworkspace.code-workspace
 /.vscode
 /build-BasicGUI-Desktop_Qt_5_15_2_MinGW_64_bit-Debug
 /build
-/CMakeLists.txt.user
\ No newline at end of file
+CMakeLists.txt.user
+/BasicGUI/CMakeLists.txt.user
```
Comment (Nutek) [inline .gitignore:6]: Patterns for CMakeLists.txt.user could be replace with one `*.txt.user`

File: BasicGUI/LedRuler.cpp [hunk 2]
```diff
@@ -1,95 +1,82 @@
...
-     painter->rotate(m_rotation);
-     painter->setBrush(m_color);
-     painter->drawRect(m_pix_rect);
+void LedRuler::paint(QPainter *painter){
+
+    painter->drawPixmap(0, 0, m_map);
+    painter->rotate(m_rotation);
+    painter->setBrush(m_color);
+    painter->drawRect(m_pix_rect);
+    qDebug() << "XXXXXXX";
```
Comment (Nutek) [inline BasicGUI/LedRuler.cpp:70]: Leftover?!

File: BasicGUI/QMLSources/DataInput.qml [hunk 3]
```diff
@@ -1,39 +1,40 @@
...
-                    activeFocusOnTab: true
-                }   
-            }    
-            Text
+    RowLayout
+    {   
+        Text
+        {
+            id: label
+            text: " "
```
Comment (Nutek) [inline BasicGUI/QMLSources/DataInput.qml:17]: Do you really need `" "` to display?

File: BasicGUI/QMLSources/main.qml [hunk 4]
```diff
@@ -2,87 +2,98 @@ import QtQuick 2.15
...
-                        DataInput{ id: l_x; value: "50"; label: "x position"}
-                        DataInput{ id: l_y; value: "50"; label: "y position"}
-                        DataInput{ id: l_size; value: "100"; label: "led size"}
-                        DataInput{ id: l_rot; value: "0"; label: "led rotation"}                      
-                }
+                showMessageBox("Open file...", `Failed to load file: ${fileDialog.fileUrl}`)
             }
+            else
+            {
+                showMessageBox("Open file...", `Success to load file: ${fileDialog.fileUrl}`)
```
Comment (Nutek) [inline BasicGUI/QMLSources/main.qml:72]: I fill success of loading is not needed to be announced

File: BasicGUI/main.cpp [hunk 5]
```diff
@@ -3,54 +3,61 @@
...
-{
-    
-
+int main(int argc, char *argv[]){
+   
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
     QCoreApplication::setAttribute(Qt::AA_DisableHighDpiScaling);
 #endif
 
+    //std::cout << "-> Application started" << std::endl;
```
Comment (Nutek) [inline BasicGUI/main.cpp:43]: Not needed

**Review by Nutek — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (1killerbehindyou1) [reply .gitignore]: Resolved

**Review by Nutek — APPROVED** [review]

**Pull request merged**